storm-core/src/clj/backtype/storm/zookeeper.clj

```diff
@@ -103,11 +103,14 @@
      (catch Exception e (throw (wrap-in-runtime e))))))
 
 (defnk delete-node
```

Comment:
It silently does nothing when path is not exist, but it was throwing NoNodeException when force is false, and default is false.
Only delete-recursive calls delete-node with force = true, and any other usages are using default.

In short, it doesn't show same behavior between two.